packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart

```diff
@@ -3,12 +3,280 @@
+
+  /// The current download state.
+  ///
+  /// When the state changes, one of the [SKTransactionObserverWrapper] subclasses' observing methods should be triggered.
+  /// The developer should properly handle the downloadable content based on the state.
+  final SKDownloadState state;
+
+  /// Length of the content in bytes.
+  final int contentLength;
+
+  /// The URL string of the content.
+  final String contentURL;
+
+  /// Version of the content formatted as a series of dot-separated integers.
+  final String contentVersion;
+
+  /// The transaction ID of the transaction that is associated with the downloadable content.
+  final String transactionID;
+
+  /// The download progress, between 0.0 to 1.0.
+  final double progress;
+
+  /// The estimated time remaining for the download; if no good estimate is able to be made,
+  /// [downloadTimeUnknown] will be set to true.
+  final double timeRemaining;
+
+  /// true if [timeRemaining] cannot be estimated.
+  final bool downloadTimeUnknown;
+
+  /// The error that prevented the downloading; only available if the [transactionState] is [SKPaymentTransactionStateWrapper.failed].
+  final SKError error;
+}
+
+/// Dart wrapper around StoreKit's [NSError](https://developer.apple.com/documentation/foundation/nserror?language=objc).
+@JsonSerializable()
+class SKError {
+  SKError(
+      {@required this.code, @required this.domain, @required this.userInfo});
+
+  /// Constructs an instance of this from a key-value map of data.
+  ///
+  /// The map needs to have named string keys with values matching the names and
+  /// types of all of the members on this class.
+  /// The `map` parameter must not be null.
+  @visibleForTesting
+  factory SKError.fromJson(Map map) {
+    assert(map != null);
+    return _$SKErrorFromJson(map);
+  }
+
+  /// Error [code](https://developer.apple.com/documentation/foundation/1448136-nserror_codes) defined in the Cocoa Framework.
+  final int code;
+
+  /// Error [domain](https://developer.apple.com/documentation/foundation/nscocoaerrordomain?language=objc) defined in the Cocoa Framework.
+  final String domain;
+
+  /// A map that contains more detailed information about the error. Any key of the map must be one of the [NSErrorUserInfoKey](https://developer.apple.com/documentation/foundation/nserroruserinfokey?language=objc).
+  final Map<String, dynamic> userInfo;
+}
+
+/// Dart wrapper around StoreKit's [SKPayment](https://developer.apple.com/documentation/storekit/skpayment?language=objc).
+///
+/// Used as the parameter to initiate a payment.
+/// In general, a developer should not need to create the payment object explicitly; instead, use
+/// [SKPaymentQueueWrapper.addPayment] directly with a product identifier to initiate a payment.
+@JsonSerializable()
+class SKPaymentWrapper {
+  SKPaymentWrapper(
+      {@required this.productIdentifier,
+      @required this.applicationUsername,
+      this.requestData,
+      this.quantity = 1,
```

Comment:
I think it's worth documenting that quantity defaults to 1 if unspecified.

Reply:
This is done on the property itself, do you mean to add a dart doc here as well? Or should we just have a comment?